python/lsst/obs/base/makeRawVisitInfo.py

```diff
@@ -95,43 +106,65 @@ def setArgDict(self, md, argDict):
 
         Subclasses are expected to override this method, though the override
         may wish to call this default implementation, which:
+
         - sets exposureTime from "EXPTIME"
         - sets date by calling getDateAvg
 
-        @param[in,out] md  metadata, as an lsst.daf.base.PropertyList or PropertySet;
-            items that are used are stripped from the metadata
-            (except TIMESYS, because it may apply to more than one other keyword).
-        @param[in,out] argdict  a dict of arguments
-
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList` or `PropertySet`
+            Metadata to pull from.
+            Items that are used are stripped from the metadata (except TIMESYS,
+            because it may apply to more than one other keyword).
+        argdict : `dict`
+            dict of arguments
+
+        Notes
+        -----
         Subclasses should expand this or replace it.
         """
         argDict["exposureTime"] = self.popFloat(md, "EXPTIME")
         argDict["date"] = self.getDateAvg(md=md, exposureTime=argDict["exposureTime"])
 
     def getDateAvg(self, md, exposureTime):
-        """Return date at the middle of the exposure
-
-        @param[in,out] md  metadata, as an lsst.daf.base.PropertyList or PropertySet;
-            items that are used are stripped from the metadata
-            (except TIMESYS, because it may apply to more than one other keyword).
-        @param[in] exposureTime  exposure time (sec)
-
-        Subclasses must override. Here is a typical implementation:
-        dateObs = self.popIsoDate(md, "DATE-OBS")
-        return self.offsetDate(dateObs, 0.5*exposureTime)
+        """Return date at the middle of the exposure.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList` or `PropertySet`
+            Metadata to pull from.
+            Items that are used are stripped from the metadata (except TIMESYS,
+            because it may apply to more than one other keyword).
+        exposureTime : `float`
+            Exposure time (sec)
+
+        Notes
+        -----
+        Subclasses must override. Here is a typical implementation::
+
+            dateObs = self.popIsoDate(md, "DATE-OBS")
+            return self.offsetDate(dateObs, 0.5*exposureTime)
         """
         raise NotImplementedError()
 
     def getDarkTime(self, argDict):
-        """Get the darkTime from the DARKTIME keyword, else expTime, else NaN
+        """Get the darkTime from the DARKTIME keyword, else expTime, else NaN,
+
+        Subclasses should call this function if desired, by putting::
```

Comment:
I suggest the following, instead. It clarifies when the method should be called and keeps related text together better (ReST required adding newlines that made the existing text harder to read):

    If dark time is available then subclasses should call this method by putting the following in their `__init__` method::
        argDict['darkTime'] = self.getDarkTime(argDict)